data-pipeline/terraform/container_apps.tf

```diff
@@ -14,136 +14,50 @@ resource "azurerm_container_app_environment" "main" {
   tags = local.common-tags
 }
 
-# resource "azurerm_user_assigned_identity" "container-app" {
-#   location            = azurerm_resource_group.resource-group.location
-#   name                = "${var.environment-prefix}containerappmi"
-#   resource_group_name = azurerm_resource_group.resource-group.name
-# }
-#
-# resource "azurerm_role_assignment" "container-app" {
-#   scope                = data.azurerm_container_registry.acr.id
-#   role_definition_name = "acrpull"
-#   principal_id         = azurerm_user_assigned_identity.container-app.principal_id
-#   depends_on = [
-#     azurerm_user_assigned_identity.container-app
-#   ]
-# }
-
-resource "azurerm_container_app" "data-pipeline" {
-  name                         = "${var.environment-prefix}-ebis-data-pipeline"
-  container_app_environment_id = azurerm_container_app_environment.main.id
-  resource_group_name          = azurerm_resource_group.resource-group.name
-  revision_mode                = "Single"
-  workload_profile_name        = "Pipeline"
-
-  identity {
-    type = "SystemAssigned"
-    #     identity_ids = [azurerm_user_assigned_identity.container-app.id]
-  }
+module "container_app_default" {
+  source = "./container_app"
 
-  secret {
-    name  = "queue-connection-string"
-    value = data.azurerm_storage_account.main.primary_connection_string
-  }
+  container-app-environment-id = azurerm_container_app_environment.main.id
+  resource-group-name          = azurerm_resource_group.resource-group.name
 
-  secret {
-    name  = "db-password"
-    value = data.azurerm_key_vault_secret.core-db-password.value
-  }
+  registry-server   = data.azurerm_container_registry.acr.login_server
+  registry-username = data.azurerm_container_registry.acr.admin_username
+  image-name        = "${data.azurerm_container_registry.acr.login_server}/${var.image-name}"
 
-  secret {
-    name  = "registry-password"
-    value = data.azurerm_container_registry.acr.admin_password
-  }
+  environment        = var.environment
+  environment-prefix = var.environment-prefix
 
-  registry {
-    server               = data.azurerm_container_registry.acr.login_server
-    username             = data.azurerm_container_registry.acr.admin_username
-    password_secret_name = "registry-password"
-    #     identity = azurerm_user_assigned_identity.container-app.id
-  }
+  db-host      = data.azurerm_key_vault_secret.core-db-domain-name.value
```

Comment:
Should values like this be pulled in via `data` references in the module? Or passed in here?

This has a clearer API for the module but I need to check whether this exposes more sensitive info. in Terraform state.

Reply:
Moved most of the sensitive values to `data` sources, passing in the key-vault details and secret-names.